drivers: sensors: tdk: refactor icp201xx_channel_get

kartben opened the pull request:
Refactor channel check so that the control flow is more readable.
Any unsupported channel will return `-ENOTSUP` early (including `CHAN_ALTITUDE` in case of FPU disabled), so that the remainder of the function involving a mutex only has one clean lock/unlock pair.

Fixes zephyrproject-rtos/zephyr#90499
Fixes Coverity CID 520292

teburd reviewed (changes requested):
This isn't a bad idea to fix the branches here, but the code is actually reachable given some conditions are met.

kartben committed:
drivers: sensors: tdk: refactor icp201xx_channel_get

Refactor channel check so that the control flow is more readable.
Any unsupported channel will return -ENOTSUP early (including
CHAN_ALTITUDE in case of FPU disabled), so that the remainder of the
function involving a mutex only has one clean lock/unlock pair.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented on drivers/sensor/tdk/icp201xx/icp201xx_drv.c:
```diff
@@ -238,9 +238,6 @@ static int icp201xx_channel_get(const struct device *dev, enum sensor_channel ch
 		altitude = convertToHeight(pressure, temperature);
 		sensor_value_from_float(val, altitude);
 #endif
-	} else {
```
Right :) How about now?
Thanks for paying attention! :)

ubieda commented on drivers/sensor/tdk/icp201xx/icp201xx_drv.c:
```diff
@@ -226,8 +226,7 @@ static int icp201xx_channel_get(const struct device *dev, enum sensor_channel ch
 		icp201xx_convert_temperature(val, data->raw_temperature);
-#ifdef CONFIG_FPU
-	} else if (chan == SENSOR_CHAN_ALTITUDE) {
+	} else if (chan == SENSOR_CHAN_ALTITUDE && IS_ENABLED(CONFIG_FPU)) {
```
At this point we've already verified the Altitude channel has the FPU enabled, right? Therefore, I agree a switch-case pattern is easier to read.

MaureenHelm commented on drivers/sensor/tdk/icp201xx/icp201xx_drv.c:
```diff
@@ -217,7 +217,7 @@ static int icp201xx_channel_get(const struct device *dev, enum sensor_channel ch
 
 	if (!(chan == SENSOR_CHAN_AMBIENT_TEMP || chan == SENSOR_CHAN_PRESS ||
-	      chan == SENSOR_CHAN_ALTITUDE)) {
+	      (chan == SENSOR_CHAN_ALTITUDE && IS_ENABLED(CONFIG_FPU)))) {
```
Shouldn't this be:
```suggestion
	      (chan == SENSOR_CHAN_ALTITUDE && !IS_ENABLED(CONFIG_FPU)))) {
```

Would be nice if the driver had a Kconfig symbol to enable/disable the altitude channel and that symbol depend on the FPU being enabled. But I think that's beyond the scope of this PR.
